samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,275 @@
+            }
+        }
+
+        /// <summary>
+        /// The OAuth Prompt needs to see the Invoke Activity in order to complete the login process.
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "adaptiveCard/action")
+            {
+                if (turnContext.Activity.Value == null)
+                    return null;
+
+                JObject value = JsonConvert.DeserializeObject<JObject>(turnContext.Activity.Value.ToString());
+
+                if (value["action"] == null)
+                    return null;
+
+                JObject actiondata = JsonConvert.DeserializeObject<JObject>(value["action"].ToString());
+
+                if (actiondata["verb"] == null)
+                    return null;
+
+                //When adaptiveCard/action invoke activity from teams contains sso token
+                string verb = actiondata["verb"].ToString();
+
+                JObject authentication = null;
+
+                if (value["authentication"] != null)
+                {
+                    authentication = JsonConvert.DeserializeObject<JObject>(value["authentication"].ToString());
+                }
+
+                // When adaptiveCard/action invoke activity from teams contains 6 digit state in response to nominal sign in flow from bot.
+                string state = null;
+                if (value["state"] != null)
+                {
+                    state = value["state"].ToString();
+                }
+
+                // authToken and state are absent, handle verb
+                if (authentication == null && state == null)
+                {
+                    switch (verb)
+                    {   //when token is absent in the invoke. We can initiate SSO in response to the invoke
+                        case "initiateSSO":
+                            return await initiateSSOAsync(turnContext, cancellationToken);
+                    }
+                }
+                else
+                {
+                    return createAdaptiveCardInvokeResponseAsync(authentication, state);
+
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Authentication success.
+        /// AuthToken or state is present. Verify token/state in invoke payload and return AC response
+        /// </summary>
+        /// <param name="authentication"></param>
+        /// <param name="state"></param>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="isBasicRefresh"></param>
+        /// <param name="fileName"></param>
```

Comment:
No explanation added for these parameters

Reply:
Fixed